src/database-layer/db.ts

```diff
@@ -707,9 +714,55 @@ if (maybePg != null) {
 				`);
 			}
 		}
+
+		let listenerClient: Pg.PoolClient | null = null;
+		async function listen(
+			channel: string,
+			fn: (...args: any[]) => Promise<void>,
+		) {
+			while (listenerClient == null) {
+				try {
+					listenerClient = await pool.connect();
```

Comment:
This is not safe because you could do
```
pg.on('notification', () => {}, 'foobar');
pg.on('notification', () => {}, 'foobar');
```
and have two concurrent loops attempting to connect and overwriting `listenerClient`